arangod/RestServer/Metrics.cpp

```diff
@@ -102,8 +102,8 @@ void Counter::store(uint64_t const& n) {
 
 void Counter::toPrometheus(std::string& result) const {
   _b.push();
-  result += "\n#TYPE " + name() + " counter\n";
-  result += "#HELP " + name() + " " + help() + "\n";
+  result += "\n# HELP " + name() + " " + help() + "\n";
+  result += "# TYPE " + name() + " counter\n";
```

Comment:
Ok... why do we need to swap the order of `TYPE` and `HELP` here, and why do we need the extra space?
According to https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md#grouping-and-sorting these changes are valid, but not necessary. 
To keep our API output as close to previous ArangoDB versions as possible, I would favor reverting this change.

Reply:
Well, yes true. The definition is a bit clumsy. I just wanted to stick to the examples given there. 
Also: Reading name + definition first makes more sense from a readers perspective than reading the type first.
Additionally, libraries I've found do want that extra space for parsing. If someone is not using prom UI, and wanted to use those libraries for parsing as well, they will run into the same problems I ran into. 

So I vote for keeping those changes ( at least the extra space - but IMHO also the order of attributes)